air/src/execution/utils/resolve.rs

```diff
@@ -53,27 +55,18 @@ fn prepare_consts(arg: impl Into<JValue>, ctx: &ExecutionCtx<'_>) -> ExecutionRe
 }
 
 #[allow(clippy::unnecessary_wraps)]
-fn prepare_last_error(ctx: &ExecutionCtx<'_>) -> ExecutionResult<(JValue, Vec<SecurityTetraplet>)> {
-    let result = match &ctx.last_error {
-        Some(error) => {
-            let serialized_error = error.serialize();
-            let jvalue = JValue::String(serialized_error);
-            let tetraplets = error
-                .tetraplet
-                .clone()
-                .unwrap_or_else(|| SecurityTetraplet::literal_tetraplet(ctx.init_peer_id.clone()));
-
-            (jvalue, vec![tetraplets])
-        }
-        None => {
-            let jvalue = JValue::String(String::new());
-            let tetraplets = vec![];
-
-            (jvalue, tetraplets)
-        }
+fn prepare_last_error(
+    path: &LastErrorPath,
+    ctx: &ExecutionCtx<'_>,
+) -> ExecutionResult<(JValue, Vec<SecurityTetraplet>)> {
+    let LastErrorWithTetraplets { last_error, tetraplets } = ctx.last_error();
+    let jvalue = match path {
+        LastErrorPath::Instruction => JValue::String(last_error.instruction),
+        LastErrorPath::Message => JValue::String(last_error.msg),
+        LastErrorPath::None => serde_json::to_value(last_error).expect("the default serializer shouldn't fail"),
```

Comment:
you can do `json!(last_error)`. It panics inside, but looks nicer :D

Reply:
Yep, will do, it really looks nicer.